TestPollingDirWatcher 系テストが flaky で main の build CI が散発的に失敗する

## 概要

`internal/infra/dir_watcher_test.go` の `PollingDirWatcher` 関連テストが flaky で、main ブランチの build CI が散発的に失敗している。直近1週間で **少なくとも7回** 同一原因で main の build ジョブが落ちている。

## 失敗しているテスト

- `TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted` (dir_watcher_test.go:325)
- `TestPollingDirWatcher_RecreatesMissingDirectory` (dir_watcher_test.go:193) ※同じ race を抱えており同様に flaky 化する可能性がある

## CI での失敗例

| 日時 | run id | branch |
| --- | --- | --- |
| 2026-04-29 23:31 | 25139270867 | main |
| 2026-04-28 05:09 | 25035078755 | main |
| 2026-04-27 04:58 | 24977340128 | main |
| 2026-04-26 02:58 | 24946781627 | main |
| 2026-04-25 14:22 | 24932984437 | main |
| 2026-04-25 09:04 | 24927374509 | main |
| 2026-04-23 11:39 | 24833140234 | main |

失敗時のエラーは共通:

```
--- FAIL: TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted (0.00s)
    testing.go:1464: TempDir RemoveAll cleanup: unlinkat /tmp/TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted3820185798: directory not empty
```

## ローカル再現

```
$ go test -run TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted -count=50 ./internal/infra/
--- FAIL: TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted (0.00s)
    testing.go:1464: TempDir RemoveAll cleanup: unlinkat /tmp/Test...: directory not empty
FAIL	github.com/canpok1/vox-actor/internal/infra	0.011s
```

`-count=50` 程度で安定して再現する。

## 原因

`PollingDirWatcher` には「監視中のディレクトリが消えていたら自動で再作成する」動作がある (`dir_watcher.go:111-119`)。

```go
func (w *PollingDirWatcher) handleListError(...) bool {
    if errors.Is(err, fs.ErrNotExist) {
        if mkErr := os.MkdirAll(dir, 0o755); mkErr != nil { ... }
        ...
    }
}
```

テスト側のシーケンス:

1. `dir1 := t.TempDir()` — `t.Cleanup` で `RemoveAll(parentDir)` が登録される。
2. `defer cancel()` — context キャンセル予約。
3. テスト本体終了。
4. `defer cancel()` 実行 → context キャンセル。**ただし watcher ゴルーチンの終了は待たない**。
5. `t.TempDir` の cleanup が `RemoveAll(parentDir)` を実行。
6. **race**: その間に watcher ゴルーチンが `listFiles(dir1)` を呼び、`ErrNotExist` を観測 → `os.MkdirAll(dir1)` でディレクトリを再作成してしまう。
7. `RemoveAll(parentDir)` が `dir1` 削除済みと思って parent を `unlinkat` しようとしたところに、watcher が `dir1` を再作成 → `parent` が空ではないので `directory not empty` で失敗。

つまり **ゴルーチンのライフサイクルが test の TempDir cleanup より長く、cleanup と並行してファイルシステム操作が走るのが flakiness の根本原因**。

## 修正方針案

context キャンセル後、watcher ゴルーチンの終了（= `fileCh` / `errCh` のクローズ）を待ってから test を終わらせる。

例:

```go
ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
fileCh1, errCh1 := watcher.Watch(ctx, dir1)
fileCh2, errCh2 := watcher.Watch(ctx, dir2)
t.Cleanup(func() {
    cancel()
    // ゴルーチン終了まで drain
    for range fileCh1 {}
    for range errCh1 {}
    for range fileCh2 {}
    for range errCh2 {}
})
```

`t.Cleanup` を `t.TempDir` 登録より後に呼ぶと LIFO で先に走るため、TempDir 削除前にゴルーチンが完全停止することを保証できる。

該当箇所:
- `internal/infra/dir_watcher_test.go:325-368` (`MultipleDirsIndependentOnOneDeleted`)
- `internal/infra/dir_watcher_test.go:193-260` (`RecreatesMissingDirectory`) — 同じ自動再作成パスを通るので同様の対応が望ましい

## 受け入れ条件

- [ ] `go test -run TestPollingDirWatcher -count=200 ./internal/infra/` がローカルで安定して PASS する
- [ ] CI の build ジョブで同じ unlinkat エラーが再発しない
- [ ] `PollingDirWatcher` 本体のロジックは変更しない（テスト側の同期不足が原因のため）

---
🤖 Generated with [Claude Code](https://claude.ai/claude-code)

--- a/internal/infra/dir_watcher_test.go
+++ b/internal/infra/dir_watcher_test.go
@@ -205,9 +205,15 @@ func TestPollingDirWatcher_RecreatesMissingDirectory(t *testing.T) {
 		infra.WithPollingDirWatcherLogger(logger),
 	)
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
 
 	fileCh, errCh := watcher.Watch(ctx, dir)
+	t.Cleanup(func() {
+		cancel()
+		for range fileCh {
+		}
+		for range errCh {
+		}
+	})
 
 	// 監視ディレクトリを削除
 	if err := os.RemoveAll(dir); err != nil {
@@ -334,10 +340,20 @@ func TestPollingDirWatcher_MultipleDirsIndependentOnOneDeleted(t *testing.T) {
 		infra.WithPollingDirWatcherLogger(logger),
 	)
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
 
-	_, errCh1 := watcher.Watch(ctx, dir1)
+	fileCh1, errCh1 := watcher.Watch(ctx, dir1)
 	fileCh2, errCh2 := watcher.Watch(ctx, dir2)
+	t.Cleanup(func() {
+		cancel()
+		for range fileCh1 {
+		}
+		for range errCh1 {
+		}
+		for range fileCh2 {
+		}
+		for range errCh2 {
+		}
+	})
 
 	// dir1 のみ削除
 	if err := os.RemoveAll(dir1); err != nil {

